src/uwtools/cli/templater.py

```diff
@@ -119,10 +111,12 @@ def parse_args(args: List[str]) -> Namespace:
     parsed = parser.parse_args(args)
     if not parsed.outfile and not parsed.dry_run and not parsed.values_needed:
         msg = "You need outfile, dry_run, or values_needed to continue."
-        raise ArgumentError(None, msg)
+        print(msg, file=sys.stderr)
+        sys.exit(1)
```

Comment:
It's useful to raise exceptions deep in library code so that they can be caught and dealt with gracefully by higher-level code, but this is CLI code and there's nothing higher that can catch that exception. An end user probably won't have much use for a stack trace -- which in any case will be very shallow and not give much context -- here. The problem pertains to the command they just typed not being quite correct, so we can just tell them about that like humans.

Reply:
This is a great rule of thumb! I'm curious if you think it might be a nice addition to the _Coding Advice_ document we have floating around.